### PR TITLE
Fix for defaults for cli args.

### LINKED ISSUE #471: kubectl cloudflow: Default for config parameters is used instead of the provided arg on command line.
**Describe the bug**
kubectl cloudflow: Default vor config parameters is used instead of the provided arg on command line.

**To Reproduce**
./kubectl-cloudflow configure swiss-knife cloudflow.streamlets.spark-process.config-parameters.configurable-message=SPARKZ

**Expected behavior**
Expect "SPARKZ" to be used
Instead this is shown:
```
Default value 'spark-original' will be used for configuration parameter 'cloudflow.streamlets.spark-process.config-parameters.configurable-message'
```

### PR DESCRIPTION
**Pull request by RayRoestenburg**
Fixes #471

### CODE REVIEW

File: kubectl-cloudflow/cmd/configure_command.go [hunk 1]
```diff
@@ -23,17 +23,19 @@ func init() {
 	configureCMD.cmd = &cobra.Command{
 		Use:   "configure",
 		Short: "Configures a deployed Cloudflow application.",
-		Example: `kubectl cloudflow configure my-app mystreamlet.hostname=localhost
+		Example: `kubectl cloudflow configure my-app --conf my-config.conf
 
-TODO DOCUMENT
+Configuration files in HOCON format can be passed through with the --conf flag. All configuration files are merged in the order that they are passed through.
```
Comment (maasg) [inline kubectl-cloudflow/cmd/configure_command.go:28]: > All configuration files are merged in the order that they are passed through.
You may want to say which order takes precedence. (first wins/last wins)

File: kubectl-cloudflow/cmd/configure_command.go [hunk 2]
```diff
@@ -23,17 +23,19 @@ func init() {
...
 
-The arguments to the command consists of optionally one
-or more '[streamlet-name].[configuration-parameter]=[value]' pairs, separated by
-a space. If the key does not point to a streamlet name in the blueprint, it is used to set application level configuration values for all streamlets, 
-for instance using 'akka.loglevel=DEBUG' (assuming there is no streamlet named akka) will set the akka loglevel to DEBUG for all streamlets.
+It is also possible to pass configuration values as command line arguments, as [config-key]=value pairs separated by
```
Comment (maasg) [inline kubectl-cloudflow/cmd/configure_command.go:30]:
```suggestion
It is also possible to pass configuration values directly through the command-line as `[config-key]=value` pairs separated by
```

File: kubectl-cloudflow/cmd/deploy_command.go [hunk 3]
```diff
@@ -45,14 +45,18 @@ func init() {
...
+
+It is also possible to pass configuration values as command line arguments, as [config-key]=value pairs separated by
+a space. The [config-key] must be an absolute path to the value, exactly how it would be defined in a config file. 
+Some examples:
+
+kubectl cloudflow deploy target/swiss-knife.json cloudflow.runtimes.spark.config.spark.driver.memoryOverhead=512
+kubectl cloudflow deploy target/swiss-knife.json cloudflow.streamlets.spark-process.config-parameters.configurable-message='SPARK-OUTPUT:'
+
+
+The arguments passed with '[config-key]=[value]' pairs take precedence over the files passed through with the --conf flag.
```
Comment (maasg) [inline kubectl-cloudflow/cmd/deploy_command.go:59]:
```suggestion
The arguments passed with `[config-key]=[value]` pairs take precedence over the files passed through with the `--conf ` flag.
```

File: kubectl-cloudflow/cmd/configure_command.go (hunk 1 above)
Comment (RayRoestenburg) [reply kubectl-cloudflow/cmd/configure_command.go:28]: That is the next line? 
> All configuration files are merged in the order that they are passed through.

Comment (RayRoestenburg) [reply kubectl-cloudflow/cmd/configure_command.go:28]: Or well, I can add a line for that to be exact.

Comment (RayRoestenburg) [reply kubectl-cloudflow/cmd/configure_command.go]: Added some more text to explain this.

**Pull request merged**